cmd/kube-controller-manager/app/plugins.go

```diff
@@ -92,15 +82,10 @@ func GetDynamicPluginProber(config persistentvolumeconfig.VolumeConfiguration) v
 func ProbeExpandableVolumePlugins(config persistentvolumeconfig.VolumeConfiguration) []volume.VolumePlugin {
 	allPlugins := []volume.VolumePlugin{}
```

Comment:
I vaguely expected all the volume plugins (other than csi) to be put in the legacyprovider file

Reply:
hostpath, emtpyDir, ... ?

This PR is only about the legacy cloud provider packages which definitely will be removed as defined in one of the cloud provider extraction KEPs. 

There's a seperate CSI migration effort that will eventually replace some of these with stub drivers to CSI since we can't remove the types.

I spoke to @davidz627 @msau42 and @saad-ali about this particular aspect.

Reply:
I think we discussed this as part of the KEP but wanted to point out again that compatibility (wrt legacy in-tree plugins) will break if the changes in this PR is built with `nolegacyproviders`. CSI migration shims (https://github.com/kubernetes/kubernetes/pull/80587/files) will need to take care of ensuring that compatibility as discussed with @davidz627

Reply:
Right, this cannot be used for official releases, only for testing because it definitely breaks compatibility. Eventually the CSI migration work should be able to solve this issue though as I understand it.

Highlighting things that don't work if we deleted these provider packages is the goal. This one just happens to already have work going to fix it :upside_down_face:

Reply:
hostPath and emptyDir should stay, but should anything else?

Reply:
it's totally fine to move the other ones in a followup. This PR doesn't need to be any larger.